### PR TITLE
Misc static analysis fixes and typos

### PR DESCRIPTION
**Pull request by LeftofZen**
A bunch of small fixes I've found. Please let me know if I've made any erroneous changes

### CODE REVIEW

File: src/OpenLoco/src/Windows/ScenarioOptions.cpp [hunk 1]
```diff
@@ -953,7 +953,7 @@ namespace OpenLoco::Ui::Windows::ScenarioOptions
         // 0x0046E306
         static uint32_t getLoanSizeInCurrency()
         {
-            uint64_t loanSizeInCurrency = Economy::getInflationAdjustedCost(CompanyManager::getStartingLoanSize(), 0, 8) / 100 * 100;
+            int32_t loanSizeInCurrency = Economy::getInflationAdjustedCost(CompanyManager::getStartingLoanSize(), 0, 8) / 100 * 100;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/ScenarioOptions.cpp:956]: I'm a little wary of this change, tbh, particularly as the sign changes. Could we double-check this?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/ScenarioOptions.cpp:956]: afaik this is correct. This has been wrong for a little while.

Comment (LeftofZen) [reply src/OpenLoco/src/Windows/ScenarioOptions.cpp:956]: `getInflationAdjustedCost` returns `currency32_t` which is `int32_t`, and a `/ 100 * 100` cannot possibly make it overflow. actually I will change this to `currency32_t here as well`

**Review by duncanspumpkin — APPROVED** [review]

File: src/OpenLoco/src/Vehicles/OrderManager.cpp [hunk 2]
```diff
@@ -404,6 +404,10 @@ namespace OpenLoco::Vehicles::OrderManager
         {
             auto rawOrder = *it;
             auto orderType = rawOrder & 0x7;
+            if (orderType >= std::size(kOrderSizes))
+            {
+                throw std::out_of_range("Order type is greater than the order size table");
```
Comment (ZehMatt) [inline src/OpenLoco/src/Vehicles/OrderManager.cpp:409]: Would be better to use our custom exception type which also adds information from where its thrown. See Core/Exception.hpp.
You have to add a new type for this one.
```
using OutOfRange = Detail::ExceptionBase<struct OutOfRangeTag>;
```

Comment (LeftofZen) [reply src/OpenLoco/src/Vehicles/OrderManager.cpp:409]: I can do that. We do have a couple of existing std::out_of_range calls though, should we change those?

Comment (ZehMatt) [reply src/OpenLoco/src/Vehicles/OrderManager.cpp:409]: Yep, I've must have missed them last time somehow.

Comment (LeftofZen) [reply src/OpenLoco/src/Vehicles/OrderManager.cpp:409]: these don't exist in master - is this something ORCT2 has?

Comment (LeftofZen) [reply src/OpenLoco/src/Vehicles/OrderManager.cpp:409]: oh i see what you mean - i might leave this to another PR

**Pull request merged**